ndinterp/src/grid.rs

```diff
@@ -11,7 +11,25 @@
 //!
 use crate::interpolate::InterpolationError;
 use itertools::izip;
-use ndarray::{Array, ArrayView1, Dimension};
+use ndarray::{Array, ArrayView1, Dimension, Ix1, Ix2};
+
+/// Together with the trait [`ToDimension`] this struct allows to convert a `usize` into a
+/// `Dimension` from the `ndarray` crate.
+pub struct DimensionHelper<const D: usize>;
+
+/// See documentation of [`DimensionHelper`].
+pub trait ToDimension {
+    /// A `Dimension` type from the `ndarray` crate.
+    type Dim: Dimension;
+}
+
+impl ToDimension for DimensionHelper<1> {
+    type Dim = Ix1;
+}
+
+impl ToDimension for DimensionHelper<2> {
+    type Dim = Ix2;
+}
```

Comment:
Ok, this is not as I expected, but at least the repeated part is quite minimal!

Thanks @scarlehoff for the workaround!

I wonder if you can give a blanket implementation for `usize`, and just overwrite for `1` and `2` (i.e. saying that by default is `Ix0`). At least, we would reduce the boilerplate at call site, since all `usize` would implement the `ToDimension` trait.

Reply:
I take it back: it's not possible (yet)

https://rust-lang.github.io/rfcs/1210-impl-specialization.html